### ISSUE TITLE
Proposal: publish to npm using scopes

### ISSUE THREAD

**Issue by mroderick** [sinonjs/sinon#1652]
I think it would make sense to start using scopes when we publish things to npm.

* `sinon` => `@sinonjs/sinon`
* `nise` => `@sinonjs/nise`
* `lolex` => `@sinonjs/lolex`
* `sinon-test` => `@sinonjs/sinon-test`, `@sinonjs/test`, `@sinonjs/testcase` or some other name

**Comment by mantoni** [comment]
Yes that would be nice and consistent with GitHub. It also gives us more freedom with naming future modules 👍

**Comment by fatso83** [comment]
How do we notify people of the naming change? Most people are probably used to adding sinon to a project using `npm install sinon`. That will make them stay at an older release.

**Comment by mgred** [comment]
> How do we notify people of the naming change?

I think we should communicate as soon as possible with a note on the website/documentation and the `README.md` in all the projects that there will be a scoped package name structure from version `x.0.0`. 

> That will make them stay at an older release

Furthermore we should publish a deprecation notice on npm, e.g. CoffeScript is doing this:
https://www.npmjs.com/package/coffee-script (1.x)
https://www.npmjs.com/package/coffeescript (2.x)

When you run `npm install coffee-script`, there will be a notice saying:
` CoffeeScript on NPM has moved to "coffeescript" (no hyphen)`

I have no experience with npm but, [`npm-depricate`](https://docs.npmjs.com/cli/deprecate) seems worth a look.

> Maybe the last major version change should just be that each `require` call now outputs `sinon has moved to the @sinonjs/sinon namespace. Please change your package.json dependency accordingly.`?

:smile: I like this. When we log a message everytime sinon is `require`d then I think users quickly update their dependecies

**Comment by mroderick** [comment]
Deprecating packages is not that difficult, I've already done it when we adopted `referee`, `samsam` and `formatio`.

**Comment by mantoni** [comment]
Issue for nise: https://github.com/sinonjs/nise/issues/77

**Comment by fatso83** [comment]
This is only missing for sinon.
